### PR TITLE
Feature/risk

### PR DESCRIPTION
**Pull request by Rambatino**

### CODE REVIEW

File: CHAID/CHAID.py [hunk 1]
```diff
@@ -198,6 +202,8 @@ class CHAID(object):
         the threshold value of the number of respondents that the node must contain (default 30)
     split_titles : array-like 
         array of names for the independent variables in the data
+    is_terminal : boolean 
+        whether the node is terminal
```
Comment (xulaus) [inline CHAID/CHAID.py]: Added in the wrong place.

File: CHAID/__main__.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def main():
         config['alpha_merge'] = nspace.alpha_merge
     if nspace.min_samples:
         config['min_sample'] = nspace.min_samples
-    CHAID.from_pandas_df(df, nspace.independent_variables, nspace.dependent_variable[0], **config).print_tree()
+    print CHAID.from_pandas_df(df, nspace.independent_variables, nspace.dependent_variable[0], **config).risk()
```
Comment (xulaus) [inline CHAID/__main__.py]: Might want to make this a mode rather than the only output.

**Comment by xulaus** [comment]
Does this need to be part of the class? It seems trivial to write once you have the tree.

File: CHAID/CHAID.py [hunk 3]
```diff
@@ -367,9 +374,31 @@ def print_tree(self):
         """ prints the tree out """
         self.to_tree().show()
 
-    def predict(self):
-        """ determines which rows fall into which node """
+    def node_predictions(self):
+        """ Determines which rows fall into which node """
+        pred = np.zeros(self.data_size)
```
Comment (xulaus) [inline CHAID/CHAID.py]:
```python
return np.array([node.node_id for node in self if node.is_terminal()])
```?

File: CHAID/CHAID.py [hunk 4]
```diff
@@ -102,7 +103,9 @@ class CHAIDNode(object):
         The dependent variable set
     """
     def __init__(self, choices=None, split_variable=None, chi=0,
-                 p=0, indices=None, node_id=0, parent=None, dep_v=None):
+                 p=0, indices=None, node_id=0, parent=None, dep_v=None,
+                 is_terminal=False
```
Comment (xulaus) [inline CHAID/CHAID.py]: not sure this needs to be passed in. We can do
```
@property
def is_terminal(self):
    return self.indices is not None`
```

File: CHAID/CHAID.py [hunk 5]
```diff
@@ -367,9 +374,31 @@ def print_tree(self):
...
+        """  
+        Determines the highest frequency of 
+        categorical dependent variable in the
+        terminal node where that row fell
+        """
         pred = np.zeros(self.data_size)
         for node in self.tree_store:
-            pred[node.indices] = node.node_id
+            if node.is_terminal:
+                pred[node.indices] = sorted(node.members.items(), key=operator.itemgetter(1))[0][0]
```
Comment (xulaus) [inline CHAID/CHAID.py]: No need to sort
```
pred[node.indices] = max(node.member, key=node.member.get)[0]
```

**Comment by Rambatino** [comment]
I think it solidifes the calculation

File: CHAID/CHAID.py (hunk 4 above)
Comment (Rambatino) [inline CHAID/CHAID.py]: It used to be like that, but it's not terminal_indices, it's indices as we get the respondents in all of the nodes. So this won't work

**Pull request merged**